app/routes/api/campaign.js

```diff
@@ -8,27 +8,27 @@ router.route('/campaigns')
 
 // create new campaign
 .post((request, response) => {
-    if(request.user) {
-        campaignPromise = campaignController.saveCampaign(request);
-        campaignPromise.then(function() {
-            return campaignController.findCampaignByTitleAndUser(request.body.title, request.user.id);
-        })
-        .then(function(result) {
-            response.send(result[0]._id);
-        })
-        .catch(function(err) {console.log(err)});
+    if (request.user) {
+        campaignController.saveCampaign(request)
+            .then(function() {
+                return Campaign.find().byTitleAndUser(request.body.title, request.user.id);
+            })
+            .then(function(result) {
+                response.send(result[0]._id);
+            })
+            .catch(function(err) {console.log(err)});
     } else {
         response.status(301).render('unauthorized', {logged_in: false});
     }
 })
 
 // fetch campaign list for a particular user
 .get((request, response) => {
-    if(request.user) {
-        campaignPromise = campaignController.findCampaignsByUser(request.user.id);
-        campaignPromise.then(function(result) {
-            response.send(result);
-        });
+    if (request.user) {
+        Campaign.find().byUser(request.user.id)
```

Comment:
For these and successive calls, would it make more sense to keep it consistent with the "findById" mongoose call? That is, instead of Campaign.find().byUser(...) using Campaign.findByUser(...)